payjoin-cli/src/app/v2/mod.rs

```diff
@@ -248,64 +205,204 @@ impl App {
+                    self.check_proposal(proposal, persister).await,
+                ReceiverTypeState::MaybeInputsOwned(proposal) =>
+                    self.check_inputs_not_owned(proposal, persister).await,
+                ReceiverTypeState::MaybeInputsSeen(proposal) =>
+                    self.check_no_inputs_seen_before(proposal, persister).await,
+                ReceiverTypeState::OutputsUnknown(proposal) =>
+                    self.identify_receiver_outputs(proposal, persister).await,
+                ReceiverTypeState::WantsOutputs(proposal) =>
+                    self.commit_outputs(proposal, persister).await,
+                ReceiverTypeState::WantsInputs(proposal) =>
+                    self.contribute_inputs(proposal, persister).await,
+                ReceiverTypeState::ProvisionalProposal(proposal) =>
+                    self.finalize_proposal(proposal, persister).await,
+                ReceiverTypeState::PayjoinProposal(proposal) =>
+                    self.send_payjoin_proposal(proposal, persister).await,
+                ReceiverTypeState::Uninitialized(_) =>
+                    return Err(anyhow!("Uninitialized receiver session")),
+                _ => return Err(anyhow!("Terminal receiver session")),
+            }
+        };
+
+        match res {
+            Ok(_) => Ok(()),
+            Err(e) => {
+                let (_, session_history) = replay_receiver_event_log(persister)?;
+                let pj_uri = match session_history.pj_uri() {
+                    Some(uri) => Some(uri.extras.endpoint().clone()),
+                    None => None,
+                };
+                let ohttp_relay = self.unwrap_relay_or_else_fetch(pj_uri).await?;
+                handle_recoverable_error(&ohttp_relay, &session_history).await?;
+
+                Err(e)
+            }
+        }
+    }
+
+    #[allow(clippy::incompatible_msrv)]
+    async fn read_from_directory(
+        &self,
+        session: Receiver<WithContext>,
+        persister: &ReceiverPersister,
+    ) -> Result<()> {
+        let mut interrupt = self.interrupt.clone();
+        let receiver = tokio::select! {
+            res = self.long_poll_fallback(session, persister) => res,
+            _ = interrupt.changed() => {
+                println!("Interrupted. Call the `resume` command to resume all sessions.");
+                return Err(anyhow!("Interrupted"));
+            }
+        }?;
+        println!("Fallback transaction received. Consider broadcasting this to get paid if the Payjoin fails:");
+        println!("{}", serialize_hex(&receiver.extract_tx_to_schedule_broadcast()));
+        self.check_proposal(receiver, persister).await
+    }
+
+    async fn check_proposal(
         &self,
         proposal: Receiver<UncheckedProposal>,
-    ) -> Result<Receiver<PayjoinProposal>, Error> {
+        persister: &ReceiverPersister,
+    ) -> Result<()> {
         let wallet = self.wallet();
+        // Receive Check 1: Can Broadcast
```

Comment:
These comments were mostly removed. This one was left in. Either remove all of them or remove this one